File: solutions/csharp/resistor-color-trio/1/ResistorColorTrio.cs
using System.Drawing;

public enum ResistorColor {
  black = 0,
  brown = 1,
  red = 2,
  orange = 3,
  yellow = 4,
  green = 5,
  blue = 6,
  violet = 7,
  grey = 8,
  white = 9
}

public static class ResistorColorTrio {
  public static string Label(string[] colors) {
    var numberString = string.Empty;
    for (var i = 0; i < 2; i++) {
      numberString += (int)Enum.Parse<ResistorColor>(colors[i]);
    }
    numberString += string.Join("", Enumerable.Repeat("0", (int)Enum.Parse<ResistorColor>(colors[2])));
    var numberValue = long.Parse(numberString);

    if (numberValue > 1e9) {
      return (numberValue / 1e9) + " gigaohms";
    }
    else if (numberValue > 1e6) {
      return (numberValue / 1e6) + " megaohms";
    }
    else if (numberValue > 1e3) {
      return (numberValue / 1e3) + " kiloohms";
    }

    return numberValue + " ohms";
  }
}

File: solutions/python/resistor-color-trio/1/resistor_color_trio.py
bands = {
    "black": "0",
    "brown": "1",
    "red": "2",
    "orange": "3",
    "yellow": "4",
    "green": "5",
    "blue": "6",
    "violet": "7",
    "grey": "8",
    "white": "9"
}

def label(colors):
    value = int("".join([bands[colors[i]] for i in range(2)]))
    value *= 10 ** int(bands[colors[2]])
    
    prefix = ""
    if value == 0:
        return "0 ohms"
    elif value % 1000000000 == 0:
        prefix = "giga"
        value /= (10 ** 9)
    elif value % 1000000 == 0:
        prefix = "mega"
        value /= (10 ** 6)
    elif value % 1000 == 0:
        prefix = "kilo"
        value /= (10 ** 3)
        
    return f"{int(value)} {prefix}ohms"
